### PR TITLE
add the -lts and -sts aliases to dotnet

### PR DESCRIPTION
**Pull request by Plarpoon**
add the **-lts** and **-sts** aliases to the dotnet packages

Related Issue: https://github.com/NixOS/nixpkgs/issues/260915

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Plarpoon** [comment]
This is my very first PR, please if I forgot something or I did something wrong feel free to tell me, I am here to fix it!

File: pkgs/development/python-modules/pythonnet/default.nix [hunk 1]
```diff
@@ -4,7 +4,7 @@
 , pytestCheckHook
 , pycparser
 , psutil
-, dotnet-sdk
+, dotnet-sdk-sts
```
Comment (eclairevoyant) [inline pkgs/development/python-modules/pythonnet/default.nix:7]: This should be -lts, as that's what the non-prefixed version refers to

**Review by mdarocha — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -913,9 +913,13 @@ with pkgs;
   dotnet-aspnetcore_7 = dotnetCorePackages.aspnetcore_7_0;
   dotnet-aspnetcore_8 = dotnetCorePackages.aspnetcore_8_0;
 
-  dotnet-sdk = dotnetCorePackages.sdk_6_0;
-  dotnet-runtime = dotnetCorePackages.runtime_6_0;
-  dotnet-aspnetcore = dotnetCorePackages.aspnetcore_6_0;
```
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:918]: existing ones shouldn't be removed for now, even if we are to remove them this is not the way

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pythonnet/default.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/development/python-modules/pythonnet/default.nix:7]: or better, pass `dotnetCorePackager` as an arg and use the _6_0 package in the inputs list

**Review by Plarpoon — COMMENTED** [review]
added your suggestions

**Comment by Plarpoon** [comment]
I see, that is entirely my mistake, I should have followed [these](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message#commit-has-not-been-pushed-online) information!

Time to learn how to amend old commits and fix them for me!

**Comment by l3m** [comment]
Is there a general agreement that -sts is meaningful? 

Please see https://github.com/NixOS/nixpkgs/issues/260915#issuecomment-1831385042.

**Pull request closed without merging**